### ISSUE TITLE
Gateway Server CPU usage increases but does not go down

### ISSUE THREAD

**Issue by allmind75** [wso2/product-apim#9356]
### Description:

Hi. I have the following issue. 
Gateway Server CPU usage increases but does not go down..
Is the setting wrong?

Gateway - deployment.toml
```
[server]
hostname = "***"
node_id = "***"
server_role = "gateway-worker"

[transport.client]
default_max_connection_per_host = 1000
max_total_connections = 3000

[indexing]
enable = false

[user_store]
type = "database_unique_id"

[database.carbon_db]
type = "mysql"
hostname ="***"
name = ***"
port = "***"
driver = "com.mysql.cj.jdbc.Driver"
username = "***"
password = "***"
validationQuery = "SELECT 1"
pool_options.maxActive = 50
pool_options.maxWait = 30000
pool_options.validationInterval = 10000

[database.apim_db]
type = "mysql"
hostname = "***"
name = "***"
port = "***"
driver = "com.mysql.cj.jdbc.Driver"
username = "***"
password = "***"
validationQuery = "SELECT 1"
pool_options.maxActive = 50
pool_options.maxWait = 10000
pool_options.validationInterval  = 10000

[database.shared_db]
type = "mysql"
hostname = "***"
name = "***"
port = "***"
driver = "com.mysql.cj.jdbc.Driver"
username = "***"
password = "***"
validationQuery = "SELECT 1"
pool_options.maxActive = 100
pool_options.maxWait = 10000
pool_options.validationInterval  = 10000

[database.metrics_db]
type = "mysql"
hostname = "***"
name = "***"
port = "***"
driver = "com.mysql.cj.jdbc.Driver"
username = "***"
password = "***"
validationQuery = "SELECT 1"
pool_options.maxActive = 100
pool_options.maxWait = 10000
pool_options.validationInterval  = 10000

[keystore.tls]
file_name = "wso2carbon.jks"
type = "JKS"
password = "***"
alias = "wso2carbon"
key_password = "***"

[truststore]
file_name = "client-truststore.jsk"
type = "JSK"
password = "wso2carbon"

[apim.key_manager]
service_url  = "***"
username= "$ref{super_admin.username}"
password= "$ref{super_admin.password}"

[apim.throttling]
username= "$ref{super_admin.username}"
password= "$ref{super_admin.username}"
enable_unlimited_tire = true
enable_header_based_throttling = false
enable_jwt_claim_based_throttling = false
enable_query_param_based_throttling = false
throttle_decision_endpoints = [***]

[[apim.throttling.url_group]]
traffic_manager_urls=[***]
traffic_manager_auth_urls=[***]
type = "loadbalance"

[apim.analytics]
enable = true
store_api_url  = "***"
username= "$ref{super_admin.username}"
password= "$ref{super_admin.username}"

[[apim.analytics.url_group]]
analytics_url=[***]
analytics_auth_url=[***]
type = "loadbalance"

[apim.cache.gateway_token]
enable = true
expiry_time = 15

[apim.cache.resource]
enable = true

[apim.oauth_config]
remove_outbound_auth_header = true
auth_header = "Authorization"

[apim.cors]
allow_origins = "*"
allow_methods = ["GET","PUT","POST","DELETE","PATCH","OPTIONS"]
allow_headers = ["authorization","Access-Control-Allow-Origin","Content-Type","SOAPAction"]
allow_credentials = false

```

### Steps to reproduce:
1.publisher creates api
2. tests the call to the gateway, the response is 200 OK.
However, the CPU usage increases and does not go down during continuous calls.

[API Call Flow]
Client -> Nginx -> Gateway -> Endpoint API

### Affected Product Version:

3.1.0

### Environment details (with versions):
- OS: Ubuntu 18.04 LTS
- JDK : open jdk 11.0.8

**Comment by arunans23** [comment]
There is a related [issue](https://github.com/wso2/product-apim/issues/8908). High CPU usage was observed in the latest JDK versions for **HTTPS** calls with a `Connection:close` header. Try downgrading your JDK version to one of the below version.

Versions prior to AdoptOpenJDK (build 11.0.8+10)
Oracle JDK 11.0.1
Versions prior to Oracle JDK 1.8.0_261

**Comment by isharac** [comment]
As @arunans23 pointed above this issue is already fixed with https://github.com/wso2/product-apim/issues/8908.
And we have fixed the same in APIM 3.0.0 and APIM 3.1.0 with wum updates.
Hence closing. Please reopen if you have further concerns.

**Comment by allmind75** [comment]
@arunans23 @isharac 
Thanks!
I'll  try downgrading JDK version.
Additional issues after testing, I'll reopen it.
